File: crates/dewasm-backend-go/units/wasi/fd_filestat_get.go
// requires: memory/init, wasi/pack_filestat
func (w *WASI) wasi_fd_filestat_get(fd, bufPtr uint32) uint32 {
    entry, present := w.fds[fd]
    if !present {
        return wasiBadf
    }
    var fi os.FileInfo
    var err error
    if d, ok := entry.(*wasiDir); ok {
        fi, err = os.Stat(d.hostPath)
    } else if f, ok := entry.(*os.File); ok {
        fi, err = f.Stat()
    }
    if err != nil || fi == nil {
        return wasiIo
    }
    w.memory.init(uint64(bufPtr), w.pack_filestat(fi), 0, 64)
    return wasiOk
}

File: crates/dewasm-backend-python/units/wasi/fd_filestat_get.py
# requires: memory/init, wasi/pack_filestat
def wasi_fd_filestat_get(self, fd, buf_ptr):
    entry = self.fds.get(fd)
    if entry is None:
        return self.ERRNO_BADF
    try:
        if isinstance(entry, self.WasiDir):
            st = os.stat(entry.host_path)
        else:
            st = os.fstat(entry.fileno())
    except OSError:
        return self.ERRNO_IO
    self.memory.init(buf_ptr, self.pack_filestat(st), 0, 64)
    return self.ERRNO_SUCCESS
